wikiclass/feature_lists/wikidatawiki.py

```diff
@@ -21,6 +39,137 @@ class items:
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if a language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if(len(value[1]) == 2): 
+			result_set.append(value[0])
+			
+	return len(result_set)
+
+def _process_important_translations(item_labels, item_descriptions):
+	item_labels_dict = item_labels
+	item_desc_dict = item_descriptions
+	combined_dict = {}
+	result_set = []
+	
+	#merging item label dictionary and item description dictionary
+	for key in (item_labels_dict.keys() | item_desc_dict.keys()):
+		if key in item_labels_dict: combined_dict.setdefault(key, []).append(item_labels_dict[key])
+		if key in item_desc_dict: combined_dict.setdefault(key, []).append(item_desc_dict[key])
+	
+	#if an important language consists of both item label and item description exist, add the language into result_set
+	for value in combined_dict.items():
+		if((len(value[1]) == 2) and (value[0] in IMPORTANT_LANG_CODES)):
+			result_set.append(value[0])
+
+	return len(result_set)/8
+
+item = Revision_Datasource(name, revision_oriented.revision).item
+"A `~pywikibase.Item` for the Wikibase content"
+
+unique_sources = Feature(name + ".unique_sources", _process_unique_sources, depends_on=[item], returns=int)
+"`int` : A count of unique sources in the revision" 
+
+item_labels = Revision_Datasource(name, revision_oriented.revision).labels
+item_descriptions = Revision_Datasource(name, revision_oriented.revision).descriptions
+complete_translations = Feature(name + ".complete_translations", _process_complete_translations, depends_on=[item_labels, item_descriptions], returns=int)
+"`int` :A count of completed translations (a pair of completed label and description) in the revision" 
+
+complete_important_translations = Feature(name + ".complete_important_translations", _process_important_translations, depends_on=[item_labels, item_descriptions], returns=float)
+"`float` : A ratio of completed important translations (a pair of completed label and description) in the revision"
+
+all_sources_datasource = Datasource(name + ".all_sources", _process_all_sources, depends_on=[item])
+all_sources = aggregators.len(all_sources_datasource)
+"`int` : A count of all sources in the revision" 
+
+all_wikimedia_sources_datasource = Datasource(name + ".all_wikimedia_sources", _process_wikimedia_sources, depends_on=[item])
+all_wikimedia_sources = aggregators.len(all_wikimedia_sources_datasource)
+"`int` : A count of all sources which come from Wikimedia projects in the revision" 
+
+all_external_sources = modifiers.sub(all_sources, all_wikimedia_sources)
+"A count of all sources which do not come from Wikimedia projects in the revision" 
+
+external_sources_ratio = all_external_sources / modifiers.max (Revision_Feature(name, Revision_Datasource(name, revision_oriented.revision)).sources, 1)
```

Comment:
@halfak : I wonder if Revision_Feature(name, Revision_Datasource(name, revision_oriented.revision) seems quite long for you.